### PR TITLE
operator: make Protocol parameters positional-only

### PR DESCRIPTION
**Pull request by JelleZijlstra**
pyanalyze checks parameter names for protocols strictly, so `float` didn't match these protocols.

### CODE REVIEW

File: stdlib/_operator.pyi [hunk 1]
```diff
@@ -37,16 +37,16 @@ class _SupportsPos(Protocol[_T_co]):
...
-    def __gt__(self, other: Any) -> Any: ...
+    def __gt__(self, __other: Any) -> Any: ...
 
 class _SupportsLE(Protocol):
-    def __le__(self, other: Any) -> Any: ...
+    def __le__(self, __other: Any) -> Any: ...
 
 class _SupportsGE(Protocol):
-    def __ge__(self, other: Any) -> Any: ...
+    def __ge__(self, __other: Any) -> Any: ...
```
Comment (AlexWaygood) [inline stdlib/_operator.pyi:50]: I regretted not adding an explanatory comment when I added these protocols (I don't think it's self-evident) -- while we're here, could you add them now?

```suggestion

# We get false-positive errors if e.g. `lt` does not have the same signature as `le`,
# so a broad union type is required for all four comparison methods
```

Comment (JelleZijlstra) [reply stdlib/_operator.pyi:50]: Sure, done

**Review by AlexWaygood — APPROVED** [review]

File: stdlib/_operator.pyi (hunk 1 above)
Comment (AlexWaygood) [reply stdlib/_operator.pyi:50]: Thanks! 😀

**Review by srittau — APPROVED** [review]

**Comment by Akuli** [comment]
Would it be possible to add pyanalyze to our CI?

**Comment by JelleZijlstra** [comment]
Not easily. The way it uses stubs doesn't really allow it to check a whole stub the way mypy does. A "pyanalyze primer" analogous to mypy primer could have found this, but the issue here showed up in our private code and very few open source projects use pyanalyze. I'm happy to just submit these PRs whenever I run into something.

**Pull request merged**